apacheconfig/wloader.py

```diff
@@ -0,0 +1,423 @@
+#
+# This file is part of apacheconfig software.
+#
+# Copyright (c) 2018-2019, Ilya Etingof <etingof@gmail.com>
+# License: https://github.com/etingof/apacheconfig/LICENSE.rst
+#
+import abc
+import six
+
+from apacheconfig import make_parser
+from apacheconfig import make_lexer
+
+
+def _restore_original(word):
+    """If the `word` is a Quoted string, restores it to original.
+    """
+    if getattr(word, 'is_single_quoted', False):
+        return "'%s'" % word
+    if getattr(word, 'is_double_quoted', False):
+        return '"%s"' % word
+    return word
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Node(object):
+    """Generic class containing data that represents a node in the config AST.
+    """
+
+    @abc.abstractmethod
+    def dump(self):
+        """Returns the contents of this node as it would appear in a config
+        file.
+        """
+
+    @abc.abstractproperty
+    def ast_node_type(self):
+        """Returns object typestring as defined by the apacheconfig parser.
+        """
+
+    @abc.abstractproperty
+    def whitespace(self):
+        """Returns preceding or trailing whitespace for this node as a string.
+       """
+
+    @abc.abstractmethod
+    @whitespace.setter
+    def whitespace(self, value):
+        """Set preceding or trailing whitespace for this node.
+
+        Args:
+            value (str): value to set whitespace to.
+        """
+
+
+class ContentsNode(Node):
```

Comment:
How much `ContentsNode` is different from `BlockNode`? I can imagine, structurally, having `BlockNode` everywhere (root, intermediate nodes) and `ItemNode` & `BlockNode` in it.

My understanding is that with `ContentsNode` we will have them scoped like:

`ContentsNode` (root) -> `BlockNode` -> `ContentsNode` -> `BlockNode` -> `ItemNode`

So I am wondering if we can always treat `BlockNode` as a "container", where `ItemNode` would always be a leaf scalar e.g.:

`BlockNode` (root) -> `BlockNode` -> `ItemNode`

WDYT?